src/utils/requestNavigationHandlers.ts

```diff
@@ -77,14 +83,151 @@ export const handleDappBrowserConnectionPrompt = (dappData: DappConnectionData):
   });
 };
 
-export const handleDappBrowserRequest = async (request: Omit<RequestData, 'displayDetails'>): Promise<string | Error> => {
+const findWalletForAddress = async (address: string) => {
   const { wallets } = store.getState().wallets;
-  const selectedWallet = findWalletWithAccount(wallets!, request.address);
-  const isReadOnlyWallet = selectedWallet!.type === walletTypes.readOnly;
+  const selectedWallet = findWalletWithAccount(wallets!, address);
+  if (!selectedWallet) {
+    return Promise.reject(new Error('Wallet not found'));
+  }
+
+  const isReadOnlyWallet = selectedWallet.type === walletTypes.readOnly;
   if (isReadOnlyWallet && !enableActionsOnReadOnlyWallet) {
     watchingAlert();
     return Promise.reject(new Error('This wallet is read-only.'));
   }
+
+  return selectedWallet;
+};
+
+interface HandleMobileWalletProtocolRequestProps
+  extends Omit<ReturnType<typeof useMobileWalletProtocolHost>, 'message' | 'handleRequestUrl' | 'sendFailureToClient'> {
+  request: RequestMessage;
+}
+
```

Comment:
@DanielSinclair MWP related work starts here. Mostly left TODO comments where I'm not sure what goes there.